src/AlertsAggregator.py

```diff
@@ -0,0 +1,222 @@
+from datetime import time, datetime, timedelta
+import requests
+GEONAMES_URL = "http://api.geonames.org/searchJSON"
+
+
+class AlertsAggregator:
+    def __init__(self, df):
+        self.df = df
+
+    # This func get dataframe and return a set of all the settlement in df
+    def return_data_list(self):
+        items_lst = self.df['data'].tolist()
+        items_lst_no_duplications = set(items_lst)
+        return items_lst_no_duplications
+
+
+    # This function get dataframe and settlement_name and return a list of all the settlement in the df that settlement_name is insid them
+    def create_user_settl_lst(self, settlement: str):
+        given_lst = self.return_data_list()
+        settl_lst = []
+        for item in given_lst:
+            if settlement.lower() == item or f"{settlement.lower()}, " in item:
+                return [item]
+            elif f"{settlement.lower()} " in item  or f" {settlement.lower()}" in item or settlement.lower() in item :
+                if item not in settl_lst:
+                    settl_lst.append(item)
+        if settl_lst != []:
+            raise WrongSettlementException(f"Incorrect settlement, please try again input one of this options: {settl_lst}")
+        return settl_lst
+
+    
+    # This function use geonames api and return True if the user settlement input is actually exist else returns False
+    def does_the_settlement_exist(self, settlement: str):
+        params = {
+                  "name":settlement,
+                  "country": "IL",
+                  "maxRows":1, 
+                  "username":"roit1",
+                  "fcode":"PPL" 
+                }
+        print(params)
+        response = requests.get(GEONAMES_URL, params=params)
+        if response.status_code == 200:    
+            if response.json()['totalResultsCount'] == 0:
+                print(response.text)
+                return False
+            else:
+                print(response.text)
+                return True
+        else:
+            raise requests.exceptions.RequestException("An error occure while try to use get request from GeoNames api")
+
+
+    # This function display the distribution of alarms in a specific settlement
+    def display_dist(self, settlement: str):
+        try:
+            settlement_lst= self.create_user_settl_lst(settlement)
+            is_settlement_exist = self.does_the_settlement_exist(settlement)
+            if settlement_lst != []:
+                # Filtered the df by the given settlement lst
+                filtered_df = self.df[self.df['data'].isin(settlement_lst)]
+                
+                # Access the 'time' column of the DataFrame
+                filtered_df['time'] = pd.to_datetime(filtered_df['time'])
+                filtered_df['hours'] = filtered_df['time'].dt.strftime("%H")
+                hour_lst = ["00","01","02","03","04","05","06","07","08","09","10","11","12","13","14","15","16","17","18","19","20","21","22","23"]
```

Comment:
what the fuck is that??
use list comprehension to create that list.

and dont shorten names !! (hour_list and not hour_lst !)

Reply:
and dont shorten names !! (hour_lists and not hour_lst !)